docker/tarfile/archive_transport.go

```diff
@@ -0,0 +1,154 @@
+package tarfile
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/containers/image/docker/policyconfiguration"
+	"github.com/containers/image/docker/reference"
+	ctrImage "github.com/containers/image/image"
+	"github.com/containers/image/types"
+)
+
+// Transport is an ImageTransport for Docker registry-hosted images.
+var Transport = tarfileTransport{}
+
+type tarfileTransport struct{}
+
+func (t tarfileTransport) Name() string {
+	return "docker-archive"
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an ImageReference.
+func (t tarfileTransport) ParseReference(reference string) (types.ImageReference, error) {
+	return ParseReference(reference)
+}
+
+// ValidatePolicyConfigurationScope checks that scope is a valid name for a signature.PolicyTransportScopes keys
+// (i.e. a valid PolicyConfigurationIdentity() or PolicyConfigurationNamespaces() return value).
+// It is acceptable to allow an invalid value which will never be matched, it can "only" cause user confusion.
+// scope passed to this function will not be "", that value is always allowed.
+func (t tarfileTransport) ValidatePolicyConfigurationScope(scope string) error {
+	// FIXME? We could be verifying the various character set and length restrictions
+	// from docker/distribution/reference.regexp.go, but other than that there
+	// are few semantically invalid strings.
+	return nil
+}
+
+// tarfileReference is an ImageReference for Docker images.
+type tarfileReference struct {
+	ref  reference.NamedTagged
+	path string
+	tag  string
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an Docker ImageReference.
+func ParseReference(refString string) (types.ImageReference, error) {
+	parts := strings.SplitN(refString, ":", 2)
+	if len(parts) < 2 {
+		return nil, fmt.Errorf("Docker-archive reference %s isn't of the form <path>:<tag>", refString)
+	}
+
+	path := parts[0]
+
+	ref, err := reference.ParseNamed(parts[1])
+	if err != nil {
+		return nil, err
+	}
+	ref = reference.WithDefaultTag(ref)
+	refTagged := ref.(reference.NamedTagged)
+
+	return tarfileReference{
+		ref:  refTagged,
+		path: path,
+		tag:  fmt.Sprintf("%s:%s", refTagged.Name(), refTagged.Tag()),
```

Comment:
Does it make sense to require a tag/Docker reference even when specifying a source, which completely ignores it?